sale_service_project/models/project.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+# (c) 2015 Antiun Ingeniería S.L. - Sergio Teruel
+# (c) 2015 Antiun Ingeniería S.L. - Carlos Dauden
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from openerp import models, api, fields
+from openerp.tools.translate import _
+
+
+class ProjectProject(models.Model):
+    _inherit = 'project.project'
+
+    task_hours = fields.Float(compute='_compute_task_hours', store=True)
+
+    @api.multi
+    @api.depends('parent_id', 'child_ids', 'tasks',
+                 'tasks.work_ids', 'tasks.work_ids.hours')
+    def _compute_task_hours(self):
+        for project in self:
+            project.task_hours = sum(project.tasks.mapped('work_ids.hours'))
+
+
+class ProjectTask(models.Model):
+    _inherit = 'project.task'
+
+    project_parent_id = fields.Many2one(
+        related='project_id.parent_id',
+        store=True,
+        string='Project Parent')
+    sale_line_id = fields.Many2one(index=True)
+    analytic_line_ids = fields.One2many(comodel_name='account.analytic.line',
+                                        compute='_compute_analytic_line_ids')
+    invoice_ids = fields.One2many(comodel_name='account.invoice',
+                                  compute='_compute_analytic_line_ids')
+    all_invoiced = fields.Boolean(compute='_compute_analytic_line_ids')
+    invoice_exists = fields.Boolean(compute='_compute_analytic_line_ids')
+    is_closed = fields.Boolean(relation='stage_id.closed')
+
+    @api.multi
+    def _compute_analytic_line_ids(self):
+        for task in self:
+            all_inv = True
+            invoice_ids = []
+            lines = task.mapped('work_ids.hr_analytic_timesheet_id.line_id')
+            if 'analytic_line_id' in task.material_ids._all_columns:
```

Comment:
Why this? Is it not better just to depend on module `project_task_materials_stock`?